### DISCUSSION TITLE
How to create a partition table based on date in postgres in sqlalchemy

### DISCUSSION THREAD

**Discussion by zch-cc** [sqlalchemy/sqlalchemy#10202] [Usage Questions] [answered]
Hi team, I am new to sqlalchemy. Could you please help me on how to create a partition table based on date daily in postgres in sqlalchemy? Is sqlalchemy automatically create partition that does not exist?

**Comment by zzzeek** [comment, marked as answer]
hi -

SQLAlchemy doesn't do any of these things automatically.    to create a new table you have to emit SQL that generates the table, so if you wanted to make a new table every day, you would make a cron job that runs a python script which emits the desired CREATE TABLE statement.  

We have an example for PG partitions illustrating how to map individual classes to PG partitions at https://github.com/sqlalchemy/sqlalchemy/wiki/PostgreSQLInheritance.    You can see that to create the INHERITS partitions, it just uses [exec_driver_sql](https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.Connection.exec_driver_sql) to emit PG commands directly.    

for dates, you likely wouldn't want to make a new subclass, just use whatever class you are mapping to the table, and then for the trigger, it would know how to select the appropriate partition table for the date data in the row.     

googling I also found this: https://github.com/pgpartman/pg_partman  which may be a way to do this already.
